Sources/Turf/Geometry.swift

```diff
@@ -4,21 +4,129 @@ import CoreLocation
 #endif
 
 
-public enum GeometryType: String {
+public enum GeometryType: String, Codable, CaseIterable {
     case Point
     case LineString
     case Polygon
     case MultiPoint
     case MultiLineString
     case MultiPolygon
+    case GeometryCollection
     
-    static let allValues: [GeometryType] = [.Point, .LineString, .Polygon, .MultiPoint, .MultiLineString, .MultiPolygon]
+//    static let allValues: [GeometryType] = [.Point, .LineString, .Polygon, .MultiPoint, .MultiLineString, .MultiPolygon]
 }
 
-public struct Geometry: Codable {
-    public var type: String
+public enum _Geometry {
+    private enum CodingKeys: String, CodingKey {
+        case type
+        case coordinates
+        case geometries
+    }
+    
+    case Point(coordinates: CLLocationCoordinate2D)
+    case LineString(coordinates: [CLLocationCoordinate2D])
+    case Polygon(coordinates: [[CLLocationCoordinate2D]])
+    case MultiPoint(coordinates: [CLLocationCoordinate2D])
+    case MultiLineString(coordinates: [[CLLocationCoordinate2D]])
+    case MultiPolygon(coordinates: [[[CLLocationCoordinate2D]]])
+    case GeometryCollection(geometries: [_Geometry])
+    
+    public var type: GeometryType {
```

Comment:
Having all these cases combined in a single `Enum` will lead to a lot of switch cases in the code... does not seem pretty to me. Am I missing something?